src/Io/Transaction.php

```diff
@@ -242,4 +254,52 @@ private function progress($name, array $args = array())
 
         echo PHP_EOL;
     }
+
+    /**
+     * Add a handler (callable object) which receives the Request object and
+     * can (or not) modify it. This handler must return a request object,
+     * whether or not there was any change.
+     * 
+     * @param callable $handler
+     * @param string $identifier Name to identify this handler in the future
+     */
+    public function addRequestHandler($handler, $identifier)
+    {
+        $this->requestHandlerQueue[$identifier] = $handler;
+    }
```

Comment:
This is an interesting concept! :+1: However, I wonder if this may not overcomplicate things here? See also comments above. What do you think about this?